public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -387,7 +387,7 @@ export default class LokiLanguageProvider extends LanguageProvider {
       const rangeParams = absoluteRange ? rangeToParams(absoluteRange) : {};
       const res = await this.request(url, rangeParams);
       this.labelKeys = res.slice().sort();
-      this.logLabelOptions = this.labelKeys.map((key: string) => ({ label: key, value: key, isLeaf: false }));
+      this.logLabelOptions = this.labelKeys?.map((key: string) => ({ label: key, value: key, isLeaf: false }));
```

Comment:
this check is also wrong, as it clearly never null in this context, set it to a local variable first could help. As sort never return null/undefined

Reply:
Problem is that `this.labelKeys` is defined as `labelKeys?: string[];` which I think is wrong because we assign at least empty `[]` in constructor.

Reply:
Changed `labelKeys` to be non-optional there.